library/src/main/java/com/google/android/exoplayer/text/eia608/Eia608TrackRenderer.java

```diff
@@ -334,21 +477,32 @@ private String getDisplayCaption() {
       return null;
     }
 
+    // Every styling span should be closed at end of line
+    closeOpenSpans( STYLE_PRIORITY.STYLE_PRIORITY_ALL );
+
     int endIndex = endsWithNewline ? buildLength - 1 : buildLength;
     if (captionMode != CC_MODE_ROLL_UP) {
-      return captionStringBuilder.substring(0, endIndex);
+      return new SpannableStringBuilder(captionStringBuilder.subSequence(0, endIndex));
```

Comment:
It's unclear why you're not just returning captionStringBuilder.subSequence(0, endIndex) directly, and having the return type be CharSequence. It looks like it should work fine.